hermes-api/src/main/java/pl/allegro/tech/hermes/api/MessageFilterSpecification.java

```diff
@@ -0,0 +1,46 @@
+package pl.allegro.tech.hermes.api;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonValue;
+
+import java.util.Map;
+
+public class MessageFilterSpecification {
+    private String type;
+    private Map<String, Object> spec;
+
+    public MessageFilterSpecification() {
+    }
+
+    @JsonCreator
+    public MessageFilterSpecification(Map<String, Object> spec) {
```

Comment:
Since you are using @JsonCreator for this constructor, the default one can be deleted, also declared fields in this class can be final as it is immutable class.